wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -0,0 +1,654 @@
+{
+    if(sign)
+        ca_cert_sig = sign;
+}
+
+/* inform user key                                                     */
+/* the function expects to be called from user application             */
+/* user has to create these key information by Renesas tool in advance.*/
+void tsip_inform_user_keys(
+    byte *encrypted_session_key,
+    byte *iv,
+    byte *encrypted_user_tls_key
+)
+{
+    g_user_key_info.encrypted_session_key = NULL;
+    g_user_key_info.iv = NULL;
+    g_user_key_info.encrypted_user_tls_key = NULL;
+    
+    if ( encrypted_session_key ) {
+        g_user_key_info.encrypted_session_key = encrypted_session_key;
+    }
+    if ( iv ) {
+        g_user_key_info.iv = iv;
+    }
+    if ( encrypted_user_tls_key ) {
+        g_user_key_info.encrypted_user_tls_key = encrypted_user_tls_key;
+    }
+}
+
+#ifndef NO_WOLFSSL_RENESAS_TSIP_TLS_SESSION
+/* convert def to tsip define */
+static byte _tls2tsipdef(byte cipher)
+{
+    byte def = R_TSIP_TLS_RSA_WITH_AES_128_CBC_SHA;
+    switch(cipher){
+        case l_TLS_RSA_WITH_AES_128_CBC_SHA:
+            break;
+        case l_TLS_RSA_WITH_AES_128_CBC_SHA256:
+            def = R_TSIP_TLS_RSA_WITH_AES_128_CBC_SHA256;
+            break;
+        case l_TLS_RSA_WITH_AES_256_CBC_SHA:
+            def = R_TSIP_TLS_RSA_WITH_AES_256_CBC_SHA;
+            break;
+        case l_TLS_RSA_WITH_AES_256_CBC_SHA256:
+            def = R_TSIP_TLS_RSA_WITH_AES_256_CBC_SHA256;
+            break;
+        default:break;
+    }
+    return def;
+}
+
+/* Sha1Hmac */
+int tsip_Sha1Hmac(const struct WOLFSSL *ssl, const byte *myInner, 
+            word32 innerSz, const byte *in, word32 sz, byte *digest, 
+            word32 verify)
+{
+    tsip_hmac_sha_handle_t _handle;
+    int ret;
+    
+    if ((ret = tsip_hw_lock()) != 0) {
+        WOLFSSL_MSG("hw lock failed\n");
+        return ret;
+    }
+    tsip_hmac_sha_key_index_t key_index;
+
+    if ( (ssl->options.side == WOLFSSL_CLIENT_END && !verify) ||
+         (ssl->options.side == WOLFSSL_SERVER_END &&  verify) )
+        XMEMCPY(key_index.value, ssl->keys.tsip_client_write_MAC_secret,
+                                                    sizeof(key_index.value));
+    else
+        XMEMCPY(key_index.value, ssl->keys.tsip_server_write_MAC_secret,
+                                                    sizeof(key_index.value));
+
+    ret = R_TSIP_Sha1HmacGenerateInit(&_handle, &key_index);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha1HmacGenerateUpdate(&_handle, (uint8_t*)myInner, 
+                                                        (uint32_t)innerSz);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha1HmacGenerateUpdate(&_handle, (uint8_t*)in, sz);
+    
+    if (ret == TSIP_SUCCESS)
+        ret = R_TSIP_Sha1HmacGenerateFinal(&_handle, digest);
+    
+    /* unlock hw */
+    tsip_hw_unlock();
+    
+    return ret;
+}
+
+/* Sha256Hmac */
+int tsip_Sha256Hmac(const struct WOLFSSL *ssl, const byte *myInner, 
```

Comment:
Do we need to sanitize input arguments against NULL here?

Reply:
Yes, we do. Added sanity check before anything do.